src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -1,736 +1,755 @@
+                            await LoadAll(ignoreFailures: false, cancellationToken).ConfigureAwait(false);
+                        }
+
+                        return;
+                    }
+
+                    await RefreshIndividualKeyValues(cancellationToken).ConfigureAwait(false);
+                    await RefreshKeyValueCollections(cancellationToken).ConfigureAwait(false);
+                    await RefreshKeyValueAdapters(cancellationToken).ConfigureAwait(false);
+                }
+                finally
+                {
+                    Interlocked.Exchange(ref _networkOperationsInProgress, 0);
+                }
+            }
+        }
+
+        public async Task<bool> TryRefreshAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                await RefreshAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (RequestFailedException e)
+            {
+                if (IsAuthenticationError(e))
+                {
+                    _logger?.LogWarning(e, LoggingConstants.RefreshFailedDueToAuthenticationError);
+                }
+                else
+                {
+                    _logger?.LogWarning(e, LoggingConstants.RefreshFailedError);
+                }
+
+                return false;
+            }
+            catch (AggregateException e) when (e?.InnerExceptions?.All(e => e is RequestFailedException) ?? false)
+            {
+                if (IsAuthenticationError(e))
+                {
+                    _logger?.LogWarning(e, LoggingConstants.RefreshFailedDueToAuthenticationError);
+                }
+                else
+                {
+                    _logger?.LogWarning(e, LoggingConstants.RefreshFailedError);
+                }
+
+                return false;
+            }
+            catch (KeyVaultReferenceException e)
+            {
+                _logger?.LogWarning(e, LoggingConstants.RefreshFailedDueToKeyVaultError);
+                return false;
+            }
+            catch (OperationCanceledException)
+            {
+                _logger?.LogWarning(LoggingConstants.RefreshCanceledError);
+                return false;
+            }
+
+            return true;
+        }
+
+        public void SetDirty(TimeSpan? maxDelay)
+        {
+            DateTimeOffset cacheExpires = AddRandomDelay(DateTimeOffset.UtcNow, maxDelay ?? DefaultMaxSetDirtyDelay);
+
+            foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers)
+            {
+                changeWatcher.CacheExpires = cacheExpires;
+            }
+
+            foreach (KeyValueWatcher changeWatcher in _options.MultiKeyWatchers)
+            {
+                changeWatcher.CacheExpires = cacheExpires;
+            }
+        }
+
+        public void ProcessPushNotification(PushNotification notification, TimeSpan? maxDelay)
```

Comment:
>notification [](http://example.com/codeflow?start=61&length=12)

Nit: can you rename the argument to `pushNotification` for consistency between this file, AzureAppConfigurationRefresher.cs and IConfigurationRefresher.cs?

Reply:
Still need to address this.